frontend/pages/labels/ManageLabelsPage/LabelsTable/LabelsTableConfig.tsx

```diff
@@ -0,0 +1,169 @@
+  accessor: string;
+  Cell:
+    | ((props: ICellProps) => JSX.Element)
+    | ((props: IDropdownCellProps) => JSX.Element);
+  disableHidden?: boolean;
+  disableSortBy?: boolean;
+  sortType?: string;
+}
+
+const generateActionDropdownOptions = (
+  currentUser: IUser,
+  label: ILabel
+): IDropdownOption[] => {
+  const options: IDropdownOption[] = [
+    {
+      label: "View all hosts",
+      disabled: false,
+      value: "view_hosts",
+    },
+  ];
+
+  const hasGlobalWritePermission =
+    isGlobalAdmin(currentUser) || isGlobalMaintainer(currentUser);
+
+  const hasLabelAuthorWritePermission =
+    isAnyTeamMaintainerOrTeamAdmin(currentUser) &&
+    label.author_id === currentUser.id;
+
+  if (hasGlobalWritePermission || hasLabelAuthorWritePermission) {
+    options.push(
+      {
+        label: "Edit",
+        disabled: false,
+        value: "edit",
+      },
+      {
+        label: "Delete",
+        disabled: false,
+        value: "delete",
+      }
+    );
+  }
+
+  return options;
+};
+
+const generateTableHeaders = (
+  currentUser: IUser,
+  onClickAction: (action: string, label: ILabel) => void
+): IDataColumn[] => {
+  return [
+    {
+      title: "Name",
+      Header: (cellProps) => (
+        <HeaderCell
+          value={cellProps.column.title}
+          isSortedDesc={cellProps.column.isSortedDesc}
+        />
+      ),
+      accessor: "name",
+      disableSortBy: false,
+      Cell: (cellProps: ICellProps) => (
+        <TextCell value={cellProps.cell.value} />
+      ),
+    },
+    {
+      title: "Description",
+      Header: (cellProps) => (
+        <HeaderCell
+          value={cellProps.column.title}
+          isSortedDesc={cellProps.column.isSortedDesc}
+        />
+      ),
+      accessor: "description",
+      Cell: (cellProps: ICellProps) => (
+        <TextCell value={cellProps.cell.value || ""} />
+      ),
+    },
+    {
+      title: "Type",
+      Header: (cellProps) => (
+        <HeaderCell
+          value={cellProps.column.title}
+          isSortedDesc={cellProps.column.isSortedDesc}
+        />
+      ),
+      accessor: "label_membership_type",
+      Cell: (cellProps: ICellProps) => (
+        <TextCell value={capitalize(cellProps.cell.value)} />
+      ),
+    },
+    {
+      title: "Actions",
+      Header: "",
+      disableSortBy: true,
+      accessor: "actions",
+      Cell: (cellProps: IDropdownCellProps) => {
+        const label = cellProps.row.original;
+        const dropdownOptions = generateActionDropdownOptions(
+          currentUser,
+          label
+        );
+        return (
+          <ViewAllHostsLink
+            rowHover
+            noLink
+            excludeChevron
+            customContent={
+              <ActionsDropdown
+                options={dropdownOptions}
+                onChange={(value: string) => onClickAction(value, label)}
+                placeholder="Actions"
+              />
+            }
+          />
```

Comment:
This renders with big text:
<img width="461" height="283" alt="image" src="https://github.com/user-attachments/assets/414961f4-2318-46b9-9dbf-36b83b18cf29" />
vs. e.g. on the users table
<img width="617" height="306" alt="image" src="https://github.com/user-attachments/assets/8c5b3400-4a85-46e1-b84f-464f85a85dcf" />